src/app/features/terms-of-use/terms-of-use.page.ts

```diff
@@ -31,11 +32,14 @@ export class TermsOfUsePage {
       .pipe(
         tap(event => {
           const postMessageEvent = event as MessageEvent;
-          if (postMessageEvent.data === 'iframe-on-load') {
-            this.iframeLoaded$.next(true);
-          }
-          if (postMessageEvent.data === 'iframeBackButtonClicked') {
-            this.navController.back();
+          const data = postMessageEvent.data as BubbleToIonicPostMessage;
+          switch (data) {
+            case BubbleToIonicPostMessage.IFRAME_ON_LOAD:
+              this.iframeLoaded$.next(true);
+              break;
+            case BubbleToIonicPostMessage.IFRAME_BACK_BUTTON_CLICKED:
+              this.navController.back();
+              break;
           }
         }),
         untilDestroyed(this)
```

Comment:
The code changes seem fine. Here are a few suggestions for improvement:
**ChatGPT response for** `git show a9e4232dfd700d07b6a61ec73910927941b129ae src/app/features/terms-of-use/terms-of-use.page.ts `

1.  Use more descriptive variable names instead of generic names like `data` or `event`.
2.  Consider using a default case in the switch statement to handle unexpected values.
3.  If `BubbleToIonicPostMessage` is only used in this file, consider importing it with a relative path instead of a full path.

Reply:
@tammyyang:
1. opinionated
2. already created asana task (see [comment](https://github.com/numbersprotocol/capture-lite/pull/2589/commits/a9e4232dfd700d07b6a61ec73910927941b129ae#r1115325035))
3. opinionated